singhsanket143/Project-IDX-Clone
# [Fix]: Environment Variable Validation with Zod

#### Description
This PR introduces validation for environment variables using **Zod**.
The changes ensure that environment variables are validated before being accessed, providing better error handling and fallback mechanisms.

#5 fix/issue .env variables validation.

#### Changes
- See the root dir of both `backend` and `frontend` where you find `.env.example` file.
- Added Zod validation to enforce correct types for environment variables.
- Replaced previous error handling with more robust checks for missing or invalid environment variables.
- Introduced default values for non-critical environment variables to ensure the app runs smoothly.

#### How to Test
1. Make sure the `.env` file includes the following variables: `MAIN_PORT`, `TERMINAL_SERVER_PORT`, `REACT_PROJECT_COMMAND`.
2. Run the backend server with the following environment variables:
   ```bash
   MAIN_PORT=3000
   TERMINAL_SERVER_PORT=4000
   REACT_PROJECT_COMMAND= npm create vite@latest my-react-app -- --template react
3. `Note` : MAIN_PORT for index.js file and editor socket and ` TERMINAL_SERVER_PORT` for ws terminal and TerminalServer.js file.
